mne/time_frequency/_stockwell.py

```diff
@@ -248,21 +250,51 @@ def tfr_stockwell(inst, fmin=None, fmax=None, n_fft=None,
     -----
     .. versionadded:: 0.9.0
     """
-    # verbose dec is used b/c subfunctions are verbose
-    data = _get_data(inst, return_itc)
-    picks = _pick_data_channels(inst.info)
-    info = pick_info(inst.info, picks)
-    data = data[:, picks, :]
+    from ..source_estimate import _BaseSourceEstimate
+
     n_jobs = check_n_jobs(n_jobs)
-    power, itc, freqs = tfr_array_stockwell(data, sfreq=info['sfreq'],
+
+    if isinstance(inst, list) or isgenerator(inst):
+        for idx, obj in enumerate(inst):
+            if idx == 0:
+                data = _get_data(obj, return_itc=False)  # don't provoke errors
+                inst = obj  # overwrite this to determine the object type later
+            else:
+                # data is simply concatenated. This should be changed later.
+                data = np.concatenate((data, _get_data(obj, return_itc=False)),
+                                      axis=0)
```

Comment:
this is going to blow up the memory no?

Reply:
Yes! that's the next things I'll work on!

Reply:
it's changed now